.github/workflows/kafka-test.yml

```diff
@@ -0,0 +1,64 @@
+name: Integration tests - Kafka
+
+on: push
+
+jobs:
+  # tests that good messages go to the good topic in Kafka
+  kafka_success_good_topic:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: coursier/cache-action@v3
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Publish Docker image
+        run: sbt 'project kafka; set Docker / version := "0.0.0"' docker:publishLocal
+      - name: Create integration resources
+        run: |
+          (cd integration && docker-compose -f ./docker-compose.yml up -d)
+          sleep 10
+      - name: Run collector
+        run: |
+          docker run --network="integration_net1" --name collector -d -v "$PWD"/.github/workflows/kafka-collector-config:/snowplow/config -p 12345:12345 snowplow/scala-stream-collector-kafka:0.0.0 --config /snowplow/config/config_good.hocon
+          sleep 15
+      - name: Send good events
+        run: |
+          while read p; do
+            curl -X POST -i http://0.0.0.0:12345/com.snowplowanalytics.snowplow/tp2  -d "$p" 2>&1 | grep -q 'Set-Cookie'
+          done <"$PWD"/.github/workflows/integration_tests/data.txt
+        shell: bash
+      - name: Check number of events in the good topic
+        run: docker exec -w /bin broker kafka-run-class.sh kafka.tools.GetOffsetShell --broker-list 0.0.0.0:9092 --topic good | grep 'good:0:55'
+      - name: Check no events got to the bad topic
+        run: docker exec -w /bin broker kafka-run-class.sh kafka.tools.GetOffsetShell --broker-list 0.0.0.0:9092 --topic bad | grep 'bad:0:0'
+      - name: Stop Docker containers
+        run: docker stop $(docker ps -aq)
+
+  # tests that oversized messages go to the bad topic in Kafka
+  kafka_success_bad_topic_oversized:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: coursier/cache-action@v3
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Publish Docker image
+        run: sbt 'project kafka; set Docker / version := "0.0.0"' docker:publishLocal
+      - name: Create integration resources
+        run: |
+          (cd integration && docker-compose -f ./docker-compose.yml up -d)
+          sleep 10
+      - name: Run collector
+        run: |
+          docker run --network="integration_net1" --name collector -d -v "$PWD"/.github/workflows/kafka-collector-config:/snowplow/config -p 12345:12345 snowplow/scala-stream-collector-kafka:0.0.0 --config /snowplow/config/config_good.hocon
```

Comment:
Is there any reason why not starting the collector with docker compose, along with zookeeper and Kafka?

Reply:
Yes, it seems that Kafka container does not properly register that it's been started (as in, the container is started, but Kafka is not ready). From my testing, even using `depends_on` in the docker-compose.yml file it does not wait properly for it to be ready.

Reply:
Right now I'm trying to implement a check on the port as @istreeter proposed above to also get rid of the hardcoded 10 second sleep.

Reply:
Added a check for existing topic instead.

Reply:
Kafka offer this mechanism via [`reconnect.backoff.max.ms`](https://kafka.apache.org/32/documentation/#producerconfigs_reconnect.backoff.max.ms) , it is 1 sec by default but we could set it to e.g. 10 sec and let it retry failed broker connection attempts up to 10 sec.

Reply:
if we can bump kafka-clients to 2.7 or later, we can benefit from [`socket.connection.setup.timeout.max.ms`](https://kafka.apache.org/32/documentation/#producerconfigs_socket.connection.setup.timeout.max.ms) config which could let us avoid current retry impl and let kafka-clients do it

that could let us run all in docker-compose as well